src/types/allocator.h

```diff
@@ -0,0 +1,98 @@
+#ifndef ALLOCATOR
+#define ALLOCATOR
+
+#include <atomic>
+#include <cstddef>
+#include <utility>
+#include <unistd.h>
+#include <sys/mman.h>
+#include <system_error>
+
+
+class MemoryArea {
+  private:
+    std::size_t area_size = 0;
+    void* area = nullptr;
+    int descriptor;         // file descriptor
+    unsigned int offset;    // offset in the file
+
+  public:
+    MemoryArea(std::size_t size, int descrp = -1, unsigned int offset = 0)
+    :area_size(size), descriptor(descrp), offset(offset) {
+      area = ::mmap(nullptr,                      // memory area where to allocate (nullptr if it does not matter)
```

Comment:
Will probably be changed to use circles mmap functionality, considering that we do not have an operating system that provides mmap

Reply:
Yes, allocator must be revised. I removed this class from the branch for now as it would not belong here in general as it is not a type of ARINC system.